numpy/core/src/umath/loops_unary_fp.dispatch.c.src

```diff
@@ -0,0 +1,213 @@
+ * #unroll = 4,      4,       2,       2#
+ */
+static void simd_@TYPE@_@kind@_@STYPE@_@DTYPE@
+(const void *_src, npy_intp ssrc, void *_dst, npy_intp sdst, npy_intp len)
+{
+    const npyv_lanetype_@sfx@ *src = _src;
+          npyv_lanetype_@sfx@ *dst = _dst;
+
+    const int vstep = npyv_nlanes_@sfx@;
+    const int wstep = vstep * @unroll@;
+    for (; len >= wstep; len -= wstep, src += ssrc*wstep, dst += sdst*wstep) {
+        /**begin repeat3
+         * #N  = 0, 1, 2, 3#
+         */
+        #if @unroll@ > @N@
+            #if @STYPE@ == CONTIG
+                npyv_@sfx@ v_src@N@ = npyv_load_@sfx@(src + vstep*@N@);
+            #else
+                npyv_@sfx@ v_src@N@ = npyv_loadn_@sfx@(src + ssrc*vstep*@N@, ssrc);
+            #endif
+            npyv_@sfx@ v_unary@N@ = npyv_@intr@_@sfx@(v_src@N@);
+        #endif
+        /**end repeat3**/
+        /**begin repeat3
+         * #N  = 0, 1, 2, 3#
+         */
+        #if @unroll@ > @N@
+            #if @DTYPE@ == CONTIG
+                npyv_store_@sfx@(dst + vstep*@N@, v_unary@N@);
+            #else
+                npyv_storen_@sfx@(dst + sdst*vstep*@N@, sdst, v_unary@N@);
+            #endif
+        #endif
+        /**end repeat3**/
+    }
+    for (; len > 0; len -= vstep, src += ssrc*vstep, dst += sdst*vstep) {
+    #if @STYPE@ == CONTIG
+        #if @repl_0w1@
+            npyv_@sfx@ v_src0 = npyv_load_till_@sfx@(src, len, 1);
+        #else
+            npyv_@sfx@ v_src0 = npyv_load_tillz_@sfx@(src, len);
+        #endif
+    #else
+        #if @repl_0w1@
+            npyv_@sfx@ v_src0 = npyv_loadn_till_@sfx@(src, ssrc, len, 1);
+        #else
+            npyv_@sfx@ v_src0 = npyv_loadn_tillz_@sfx@(src, ssrc, len);
+        #endif
+    #endif
+        npyv_@sfx@ v_unary0 = npyv_@intr@_@sfx@(v_src0);
+    #if @DTYPE@ == CONTIG
+        npyv_store_till_@sfx@(dst, len, v_unary0);
+    #else
+        npyv_storen_till_@sfx@(dst, sdst, len, v_unary0);
+    #endif
+    }
+    npyv_cleanup();
+}
+/**end repeat2**/
+/**end repeat1**/
+#endif // @VCHK@
+/**end repeat**/
+
+/********************************************************************************
+ ** Defining ufunc inner functions
+ ********************************************************************************/
+/**begin repeat
+ * #TYPE = FLOAT, DOUBLE#
+ * #sfx  = f32, f64#
+ * #VCHK = NPY_SIMD, NPY_SIMD_F64#
+ */
+/**begin repeat1
+ * #kind  = sqrt, absolute, square, reciprocal#
+ * #intr  = sqrt, abs,      square, recip#
+ * #clear = 0,    1,        0,      0#
+ */
+NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(@TYPE@_@kind@)
+(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
+{
+    const char *src = args[0]; char *dst = args[1];
+    const npy_intp src_step = steps[0];
+    const npy_intp dst_step = steps[1];
+    npy_intp len = dimensions[0];
+#if @VCHK@
+    const int lsize = sizeof(npyv_lanetype_@sfx@);
+    if (src_step % lsize != 0 || dst_step % lsize != 0) {
```

Comment:
Do we need to check stride alignment?

Reply:
Was added in https://github.com/numpy/numpy/pull/16666. It was required for complex data types, I dont you need it for regular floats and doubles (see https://github.com/numpy/numpy/issues/16660#issuecomment-647644684)

Reply:
thank you! I gonna replace it with an assertion